src/transformers/tokenization_utils.py

```diff
@@ -132,114 +161,302 @@ def truncate_and_pad(
+            word_index (:obj:`int`, `optional`):
+                If a batch index is provided in `batch_or_token_index`, this can be the index
+                of the word in the sequence.
 
         Returns:
-            tuple: (word start, word end) representing the first and last characters of the word
+            char_span (:obj:`CharSpan` or :obj:`List[CharSpan]`):
+                Span(s) of the associated character or characters in the string.
+                CharSpan are NamedTuple with:
+                    start: index of the first character associated to the token in the original string
+                    end: index of the character following the last character associated to the token in the original string
         """
 
         if not self._encodings:
-            raise ValueError("char_to_word_offsets() is not available when using Python based tokenizers")
-        return self[sentence].char_to_word_offsets(char)
+            raise ValueError("word_to_chars() is not available when using Python based tokenizers")
+        if word_index is not None:
+            batch_index = batch_or_word_index
+        else:
+            batch_index = 0
+            word_index = batch_or_word_index
+        return CharSpan(*(self._encodings[batch_index].word_to_chars(word_index)))
 
-    def token_to_word_offsets(self, sentence: int, index: int) -> Optional[Tuple[int, int]]:
-        """
-        Find the Offsets of the word containing the token at the given index
+    def char_to_word(self, batch_or_char_index: int, char_index: Optional[int] = None) -> int:
+        """ Get the word in the original string corresponding to a character in the original string of
+            a sequence of the batch.
+
+            Can be called as:
+                - self.char_to_word(char_index) if batch size is 1
+                - self.char_to_word(batch_index, char_index) if batch size is greater than 1
+
+            This method is particularly suited when the input sequences are provided as
+            pre-tokenized sequences (i.e. words are defined by the user). In this case it allows
+            to easily associate encoded tokens with provided tokenized words.
 
         Args:
-            sentence (int): Index of the sentence relative to the batch provided to the tokenizer
-            index (int): Index of the token to map to the original word offsets
+            batch_or_char_index (:obj:`int`):
+                Index of the sequence in the batch. If the batch only comprise one sequence,
+                this can be the index of the character in the orginal string.
+            char_index (:obj:`int`, `optional`):
+                If a batch index is provided in `batch_or_token_index`, this can be the index
+                of the character in the orginal string.
+
 
         Returns:
-            Optional[tuple]: (word start, word end) or None
+            token_index (:obj:`int` or :obj:`List[int]`):
```

Comment:
Type hint of the `char_to_word` functions then needs to be adjusted, e.g. to `Union[int, List[int]]